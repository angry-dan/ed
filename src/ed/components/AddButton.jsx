import React from "react";
import {connect} from "react-redux";
import {addComponent} from "../modules/components";

const component = ({addComponent}) => <div className="fixed-action-btn">
  <a className="btn-floating red btn-large">
    <i className="large material-icons">add</i>
  </a>
  <ul>
    <li>
      <button onClick={() => addComponent('header')} className="btn-floating red" title="Add header"><i className="material-icons">insert_chart</i></button>
    </li>
    <li>
      <button title="Add body" className="btn-floating yellow darken-1" onClick={() => addComponent('body')}><i className="material-icons">format_quote</i></button>
    </li>
  </ul>
</div>;


export default connect(undefined, {addComponent})(component);

