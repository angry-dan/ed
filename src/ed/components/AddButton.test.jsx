import React from 'react';
import {component as AddButton} from "./AddButton";
import renderer from 'react-test-renderer';


it('renders correctly', () => {
  const tree = renderer.create(<AddButton componentTypes={{
    testComponent: {description: "A test component"}
  }}/>).toJSON();
  expect(tree).toMatchSnapshot();

  // TODO click that add button and check that the onClick event get's called
  // in the way we expect.
});
