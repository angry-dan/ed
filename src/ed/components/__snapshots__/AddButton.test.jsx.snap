// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <a
    className="btn-floating red btn-large"
  >
    <i
      className="large material-icons"
    >
      add
    </i>
  </a>
  <ul>
    <li>
      <button
        className="btn-floating red"
        onClick={[Function]}
        title="Add A test component"
      >
        <i
          className="material-icons"
        >
          format_quote
        </i>
      </button>
    </li>
  </ul>
</div>
`;
